jujugui/static/gui/src/app/store/env/controller-api.js

```diff
@@ -856,15 +856,21 @@ window.yui.add('juju-controller-api', function(Y) {
           return;
         }
         if (data.error) {
-          callback(data.error, {});
+          callback([data.error], {});
           return;
         }
+        let errors = [];
         const results = data.response.results.reduce((prev, result, index) => {
           const id = ids[index];
-          prev[id] = result.error ? result.error.message : null;
+          if (result.error) {
+            prev[id] = result.error.message;
+            errors.push(result.error.message);
+          } else {
+            prev[id] = null;
+          }
           return prev;
         }, {});
-        callback(null, results);
+        callback(errors, results);
```

Comment:
`errors` here would be truthy (`[]` is truthy) so I'd prefer it to stay as null so consumers of the api could still go `if (!error) { ... }`

What do you think about: 
`callback(errors.length ? errors : null, results);`

Reply:
This is such a ridiculously small win :P If you say so...

Reply:
It's just about creating a consistent api for those who are consuming it.